backend/pkg/api/handlers/internal.go

```diff
@@ -61,18 +61,151 @@ func (h *HandlerService) InternalGetLatestState(w http.ResponseWriter, r *http.R
 // Ad Configurations
 
 func (h *HandlerService) InternalPostAdConfigurations(w http.ResponseWriter, r *http.Request) {
-	returnCreated(w, nil)
+	var v validationError
+	user, err := h.getUserBySession(r)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+	if user.UserGroup != "ADMIN" {
+		returnUnauthorized(w, errors.New("user is not an admin"))
+		return
+	}
+
+	var req types.AdConfigurationData
+	if err := v.checkBody(&req, r); err != nil {
+		handleErr(w, err)
+		return
+	}
+	key := v.checkKeyNotEmpty(req.Key)
+	if len(req.JQuerySelector) == 0 {
+		v.add("jquery_selector", "must not be empty")
+	}
+	insertMode := checkEnum[enums.AdInsertMode](&v, req.InsertMode, "insert_mode")
+	if req.RefreshInterval == 0 {
+		v.add("refresh_interval", "must be greater than 0")
+	}
+	if (req.BannerId == 0) == (req.HtmlContent == "") {
+		returnConflict(w, errors.New("provide either banner_id or html_content"))
```

Comment:
400 bad request would be the more appropriate status code. also needs to check that only one of `banner_id` or `html_content` is set